applicationset/services/repo_service.go

```diff
@@ -42,6 +45,26 @@ func NewArgoCDService(db db.ArgoDB, gitCredStore git.CredsStore, submoduleEnable
 	}
 }
 
+func (a *argoCDService) CommitSHA(ctx context.Context, repoURL string, revision string) (string, error) {
+	repo, err := a.repositoriesDB.GetRepository(ctx, repoURL)
+	if err != nil {
+		return "", fmt.Errorf("Error in GetRepository: %w", err)
+	}
+
+	gitRepoClient, err := git.NewClient(repo.Repo, repo.GetGitCreds(a.storecreds), repo.IsInsecure(), repo.IsLFSEnabled(), repo.Proxy)
+
+	if err != nil {
+		return "", err
```

Comment:
Can we format the error with additional details here? Maybe `fmt.errorf("Error creating git client for repo %s", repoUrl)`